main.go

```diff
@@ -108,10 +132,15 @@ func runAPIPolling(done chan error, url, token string, organizationIDs []string,
 		for _, organization := range organizations {
 			log.Debugf("Collecting for organization '%s'", organization.Name)
 			err := collect(&client, organization)
+			
 			if err != nil {
 				done <- err
 				return
 			}
+
+			readyMutex.Lock()
+			defer readyMutex.Unlock()
```

Comment:
Now I come to think of it. This is actually not right. This marks the exporter ready when the first organization is scraped. Not all of them.
I think we should move it outside the `for range` loop. Do you agree?

Further more, the `defer` keyword is executed when a function is returning. This is not the case here, so we need the initially proposed handling of the mutex.
```go
for {
  for _, organization := range organizations {
    // handling
  }
  readyMutex.Lock()
  ready = true
  readyMutex.Unlock()
  time.Sleep(requestInterval)
}
```

Reply:
yes, you're right - I did not notice the for loop, will fix!

Reply:
BTW what about running the scarp in parallel?

Reply:
I've not done anything for this yet as I'd like to get some usage before hand. It is an obivious performance optimization but I'm not use it really makes any usage difference. The target is a slow moving metric, so whether it's ready in 10 seconds or 1 minute is usually not a concern.

Do you think differently? With 900 projects I can see it'll take around 6 minutes. Is that a problem for you?

Reply:
It just means it will take a lot of time to start. It only depends on the effort - if it is a small fix (I guess it is not, multi-threading is always hard) - I think it worth it. Otherwise, let's ignore it and keep an issue for tracking?